src/main/java/com/example/onlinebookstore/repository/ShoppingCartRepository.java

```diff
@@ -0,0 +1,12 @@
+package com.example.onlinebookstore.repository;
+
+import com.example.onlinebookstore.model.ShoppingCart;
+import com.example.onlinebookstore.model.User;
+import java.util.Optional;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface ShoppingCartRepository extends JpaRepository<ShoppingCart, Long> {
+    Optional<ShoppingCart> findShoppingCartByUser(User user);
```

Comment:
this is weird to search by user

you might need to search by some concrete property, like email